### ISSUE TITLE
Using "Remove ANSI Color Codes" or "Parse ANSI Color Codes" removes all console lines

### ISSUE THREAD

**Issue by mertzt89** [deprint/build-tools-cpp#54]
When using "Remove ANSI Color Codes" or "Parse ANSI Color Codes" options it causes the console output to be blank instead of just removing/parsing the ANSI formatting.

OS: Windows 7
atom: 1.3.2
build-tools: 4.3.2

**Comment by mertzt89** [comment]
I didn't see #49 which may be addressing this

**Comment by deprint** [comment]
#49 is an enhancement, not a bug fix.
Do you use `pty.js` and/or `execute in shell` for your command ?
Also: Does Windows even support ANSI Codes ? Sorry, but I have most of my windows knowledge from [Wikipedia](https://en.wikipedia.org/wiki/ANSI_escape_code#Windows_and_DOS)

**Comment by mertzt89** [comment]
I use both pty.js and "execute in shell" with command "apm show build-tools"

Console output with set to "Ignore ANSI Color Codes":

```
[?25l
[0m[0K
[1G[?25h]0;npm[?25l
build-tools@4.3.2 C:\Users\mertz\github\build-tools[0K
├─┬ atom-space-pen-views@2.1.0[0K
│ ├── fuzzaldrin@2.1.0[0K
│ └─┬ space-pen@5.1.1[0K
│   ├─┬ grim@1.5.0[0K
│   │ └─┬ emissary@1.3.3[0K
│   │   ├─┬ es6-weak-map@0.1.4[0K
│   │   │ ├── d@0.1.1[0K
│   │   │ ├─┬ es5-ext@0.10.11[0K
│   │   │ │ ├── es6-iterator@2.0.0[0K
│   │   │ │ └── es6-symbol@3.0.2[0K
│   │   │ ├── es6-iterator@0.1.3[0K
│   │   │ └── es6-symbol@2.0.1[0K
│   │   ├── mixto@1.0.0[0K
│   │   └── property-accessors@1.1.3[0K
│   ├── jquery@2.1.4[0K
│   └─┬ underscore-plus@1.6.6[0K
│     └── underscore@1.6.0[0K
├─┬ fs-plus@2.8.1[0K
│ ├── async@0.2.10[0K
│ ├── mkdirp@0.3.5[0K
│ ├── rimraf@2.2.8[0K
│ └─┬ underscore-plus@1.6.6[0K
│   └── underscore@1.6.0[0K
├─┬ ptyw.js@0.3.7[0K
│ ├── extend@3.0.0[0K
│ └── nan@2.1.0[0K
├─┬ season@5.3.0[0K
│ ├─┬ cson-parser@1.0.9[0K
│ │ └── coffee-script@1.9.0[0K
│ └─┬ optimist@0.4.0[0K
│   └── wordwrap@0.0.3[0K
└── xregexp@2.0.0[0K
[0K
[0K
[1G[?25h
```

With set to "Remove ANSI Color Codes":
In lib/pipeline/output-stream.coffee i added some prints
``` coffee-script
        data = data.substr(0, m.index)
      return data

    in: (data) ->
      console.log 'before: ', data
      data = @removeAnsi data if @stream.highlighting isnt 'nh' or @stream.ansi_option is 'remove'
      console.log 'after: ', data
      return if data is ''
      @buffer += data
```
Result on output console: Blank

Output on debug console:
```
before:  [?25l
[0m[0K
[1G[?25h
after:  

before:  [?25l
[0;36mbuild-tools[0m[0K
├── [0;33m4.3.2[0m[0K
├── https://github.com/deprint/build-tools-cpp[0K
├── Build your projects in atom[0K
├── 14747 downloads[0K
└── 16 stars[0K
[0K
Run `apm install build-tools` to install this package.[0K
[0K
[0K
[1G[?25h
after:  
```

I could probably spend more time to help you debug/fix this and continue to do windows testing. I more or less wanted to make sure that you intend for it to be functional at this stage.

**Comment by deprint** [comment]
Actually I think I already found the problem (and it's not windows-specific, so I can fix it):
The ANSI Specification is quite large (at least for a simple console like mine). The package only implements the most common color codes, which are more than sufficient for building projects, and ignores all other codes.
... at least, that's what it __should__ do. 
Right now, it does not detect when a non-color code ends, so it assumes that the entire output is one large escape code and `removeAnsi` then does what it is supposed to do and removes it.
I should be able to fix it in the next couple of days :bow:

**Comment by deprint** [comment]
Fixed in `v4.3.3`.
`Remove ANSI` removes __all__ ANSI Codes.
`Parse ANSI` only removes and highlights color codes, which means that, in your example, `\x1b[0K` and others will remain visible.
